src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -143,6 +143,10 @@ private function handleArrayNode(ArrayNode $node, ClassBuilder $class, string $n
         );
         $body = $hasNormalizationClosures ? '
 /**
+ * @template T
+ * @psalm-param T $value
+ * @psalm-return (T is array ? CLASS : static)
```

Comment:
Using vendor prefixed annotations, because e.g. PhpStorm has no support for conditional types yet afaik. This allows using `@return` with something everyone understands, while providing the necessary extra details to static analysis tools.

Reply:
let's still move this after `@return` in all cases :)

Reply:
let's still move this after `@return` in all cases :)

Reply:
Let's just say I was a bit unfocused yesterday :smile: